### PR TITLE
JW-48 Fix instance IP allocation.

### PR DESCRIPTION
**Pull request by m-kostrzewa**
Instance IP and other objects are now allocated correctly in Contrail.

### CODE REVIEW

File: controller/controller_test.go [hunk 1]
```diff
@@ -245,22 +248,25 @@ var _ = Describe("Controller", func() {
...
 				Expect(err).ToNot(HaveOccurred())
 				Expect(instanceIP).ToNot(BeNil())
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
 
 				existingIP, err := types.InstanceIpByUuid(client.ApiClient, instanceIP.GetUuid())
 				Expect(err).ToNot(HaveOccurred())
 				Expect(existingIP.GetUuid()).To(Equal(instanceIP.GetUuid()))
-
-				// TODO: check if got an IP address (instance_ip_address)
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
```
Comment (mclapinski) [inline controller/controller_test.go]: You have already checked this in line 264 so looks useless to me.

**Review by mclapinski — CHANGES_REQUESTED** [review]

**Comment by m-kostrzewa** [comment]
Fix'd, LGTY? @michal-clapinski

**Review by mclapinski — APPROVED** [review]

File: controller/controller.go [hunk 2]
```diff
@@ -100,7 +105,12 @@ func (c *Controller) GetOrCreateInterface(net *types.VirtualNetwork,
 	if err != nil {
 		return nil, err
 	}
-	return iface, nil
+
+	createdIface, err := types.VirtualMachineInterfaceByName(c.ApiClient, instance.GetName())
+	if err != nil {
+		return nil, err
```
Comment (mmasztalerczuk) [inline controller/controller.go]: Maybe print error log here?

File: controller/controller.go [hunk 3]
```diff
@@ -75,7 +75,12 @@ func (c *Controller) GetOrCreateInstance(tenantName, containerId string) (*types
 	if err != nil {
 		return nil, err
 	}
-	return instance, nil
+
+	createdInstance, err := types.VirtualMachineByName(c.ApiClient, containerId)
+	if err != nil {
+		return nil, err
```
Comment (mmasztalerczuk) [inline controller/controller.go]: Add error log here.

File: controller/controller.go [hunk 4]
```diff
@@ -132,5 +142,10 @@ func (c *Controller) GetOrCreateInstanceIp(net *types.VirtualNetwork,
 	if err != nil {
 		return nil, err
 	}
-	return instIp, nil
+
+	allocatedIP, err := types.InstanceIpByUuid(c.ApiClient, instIp.GetUuid())
+	if err != nil {
+		return nil, err
```
Comment (mmasztalerczuk) [inline controller/controller.go]: Add error log here.

File: controller/controller_test.go [hunk 5]
```diff
@@ -245,22 +248,23 @@ var _ = Describe("Controller", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(instanceIP).ToNot(BeNil())
 				Expect(instanceIP.GetUuid()).To(Equal(testInstanceIP.GetUuid()))
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
```
Comment (mmasztalerczuk) [inline controller/controller_test.go]: Why are you checking not empty string? Maybe Expect(instanceIP.GetInstanceIpAddress()).To(Equal(testInstanceIP.GetInstanceIpAddress())) ?

File: controller/controller_test.go [hunk 6]
```diff
@@ -245,22 +248,23 @@ var _ = Describe("Controller", func() {
...
 		Context("when instance IP doesn't exist in Contrail", func() {
 			It("creates new instance IP", func() {
 				instanceIP, err := client.GetOrCreateInstanceIp(testNetwork, testInterface)
 				Expect(err).ToNot(HaveOccurred())
 				Expect(instanceIP).ToNot(BeNil())
+				Expect(instanceIP.GetInstanceIpAddress()).ToNot(Equal(""))
 
 				existingIP, err := types.InstanceIpByUuid(client.ApiClient, instanceIP.GetUuid())
 				Expect(err).ToNot(HaveOccurred())
 				Expect(existingIP.GetUuid()).To(Equal(instanceIP.GetUuid()))
```
Comment (mmasztalerczuk) [inline controller/controller_test.go:268]: Why did you not check ip here? Checking Uuid is enough? (if yes, why are you checking not empty string in 263 line)

Comment (m-kostrzewa) [reply controller/controller_test.go:268]: I don't know what IP will contrail allocate for us

**Review by mmasztalerczuk — APPROVED** [review]

**Pull request merged**